src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj

```diff
@@ -13,4 +13,35 @@
     <Compile Include="System.Runtime.CompilerServices.Unsafe.il" />
     <Reference Include="System.Runtime" />
   </ItemGroup>
+
+  <Target Name="AddResourcesFileToIlasm" BeforeTargets="CoreCompile" Condition="'$(OS)'=='Windows_NT'">
+    <MakeDir Directories="$(IntermediateOutputPath)Disassembly" />
+
+    <PropertyGroup>
+      <_IldasmCommand>$(ToolsDir)ilasm/ildasm.exe</_IldasmCommand>
+      <_IldasmCommand>$(_IldasmCommand) $(ContractOutputPath)/$(MSBuildProjectName).dll</_IldasmCommand>
+      <_IldasmCommand>$(_IldasmCommand) /OUT=$(IntermediateOutputPath)Disassembly/$(MSBuildProjectName).il</_IldasmCommand>
+
+      <_CvtResCommand>cvtres.exe</_CvtResCommand>
+      <_CvtResCommand>$(_CvtResCommand) /MACHINE:x86</_CvtResCommand>
+      <_CvtResCommand>$(_CvtResCommand) /OUT:$(IntermediateOutputPath)Disassembly/$(MSBuildProjectName).obj</_CvtResCommand>
+      <_CvtResCommand>$(_CvtResCommand) $(IntermediateOutputPath)Disassembly/$(MSBuildProjectName).res</_CvtResCommand>
+    </PropertyGroup>
+
+    <!-- Getting the res file by disassemblying the contract assembly -->
+    <Exec Command="$(_IldasmCommand)" ConsoleToMSBuild="true" StandardOutputImportance="Low">
+      <Output TaskParameter="ExitCode" PropertyName="_IldasmCommandExitCode" />
+    </Exec>
+    <Error Condition="'$(_IldasmCommandExitCode)' != '0'" Text="ILDasm failed while running command: &quot;$(_IldasmCommand)&quot;" />
+
+    <!-- Calling cvtres.exe which should be on the path in order to transform the resource file to an obj -->
```

Comment:
Where is cvtres going to come from?

Reply:
Good question... Why do we even need it?  Shouldn't ILDasm be emitting a format that roundtrips through ILAsm?

Reply:
We have a requirement right now saying that you have to be running on a developer command prompt in order to build corefx. That means that cvtres will be on the PATH. I know that this means we would add a dependency to VS (when we are trying to remove as much of them as possible) but we will only add this for Windows build and I will log an issue to ilasm so that they can provide another way of passing in version information without having to transform the .res file.

Reply:
> Why do we even need it? Shouldn't ILDasm be emitting a format that roundtrips through ILAsm?

Talked to @jkotas about this offline, basically, ILDasm doesn't emit the format that ILAsm would be able to understand for now so we can submit a feature request for this.